addons/project/tests/test_project_subtasks.py

```diff
@@ -242,15 +253,23 @@ def test_subtask_creation_with_form(self):
             4) check the correct nb of sub-tasks is displayed in the 'sub-tasks' stat button and on the parent task kanban card
             5) sub-tasks should be copied when the parent task is duplicated
         """
+        test_subtask_1 = self.env['project.task'].create({
+            'name': 'Test Subtask 1',
+        })
         with Form(self.task_1.with_context({'tracking_disable': True})) as task_form:
-            with task_form.child_ids.new() as subtask_form:
-                subtask_form.name = 'Test Subtask 1'
-                subtask_form.display_project_id = self.env['project.project']
+            task_form.child_ids.add(test_subtask_1)
+            task_form.child_ids[0].display_project_id = self.env['project.project']
+            task = task_form.save()
         child_subtask = self.task_1.child_ids[0]
+        test_subtask_2 = self.env['project.task'].create({
+            'name': 'Test Subtask 2',
+        })
         with Form(child_subtask.with_context(tracking_disable=True)) as subtask_form:
-            with subtask_form.child_ids.new() as subsubtask_form:
-                subsubtask_form.name = 'Test Subtask/Subtask'
-        task = task_form.save()
+            subtask_form.child_ids.add(test_subtask_2)
+            subtask_form.child_ids[0].display_project_id = self.env['project.project']
+            subtask_form.save()
```

Comment:
Same comment here.
```suggestion
```

Reply:
i'll ask it once here, not on every comments :D 
In the test there are some other occurrence of form.save() that were used. Should I remove it too ? Since those save() were used when the task_form was a many2one instead of the m2m, i dont know if it changes something...

Reply:
In fact, after the `with` statement [this method](https://github.com/odoo/odoo/blob/fe5fbbebd9f073dff4f35b0142ed96f1044e1ad1/odoo/tests/common.py#L2300-L2302) will be called and so it will do the save method.

I suggest to just change the lines in which you touched/added. 🙂